### DISCUSSION TITLE
compress a model

### DISCUSSION THREAD

**Discussion by zjd11** [deepmodeling/deepmd-kit#1133] [Q&A] [answered]
I have trained a model, then the freezed model was compressed and put into a new train to obtain a new model.
There is no problem in this stage.
But if i compress the new model, there is an error, such as "raise KeyError("The name %s refers to an Operation not in the "KeyError: "The name 'ProdEnvMatA' refers to an Operation not in the graph.""
Moreover, if i add some new data to further re-train the model, some errors will be encountered.
for example, use "dp train --init-frz-model nnp1a.pb input.json 2>err" commond ,the error is "ValueError: Requested return tensor 'o_rmat:0' not found in graph def".
use "dp train --init-model model.ckpt input.json 2>err" commond ,the error is "tensorflow.python.framework.errors_impl.NotFoundError: Restoring from checkpoint failed. This is most likely due to a Variable name or other graph key that is missing from the checkpoint. Please ensure that you have not altered the graph expected based on the checkpoint. Original error:".
what should i do?

**Comment by njzjz** [comment, marked as answer]
The new model has been compressed so you cannot compress it twice.

**Reply by njzjz** [reply]
@denghuilu maybe we can raise a `RuntimeError` for this situation?

**Reply by zjd11** [reply]
So, what should i do, if i want to further train a model based on the compressed model, because it doesn't seem to be able to train with 'dp train --init-frz-model' commond.

**Reply by denghuilu** [reply]
> So, what should i do, if i want to further train a model based on the compressed model, because it doesn't seem to be able to train with 'dp train --init-frz-model' commond.

It seems like a bug within the implementation of 'dp train --init-frz-model', we will address it soon.

**Reply by denghuilu** [reply]
> @denghuilu maybe we can raise a `RuntimeError` for this situation?

sure

**Reply by denghuilu** [reply]
> > So, what should i do, if i want to further train a model based on the compressed model, because it doesn't seem to be able to train with 'dp train --init-frz-model' commond.
> 
> It seems like a bug within the implementation of 'dp train --init-frz-model', we will address it soon.

See #1137

**Reply by denghuilu** [reply]
> > @denghuilu maybe we can raise a `RuntimeError` for this situation?
> 
> sure

see #1136
